mountpoint-s3-crt/src/s3/endpoint_resolver.rs

```diff
@@ -147,7 +147,7 @@ impl Drop for RequestContext {
 }
 
 /// [ResolvedEndpoint] for the given context using [RuleEngine] and rule set
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
`aws_endpoints_resolved_endpoint` is reference counted, so this can't derive `Clone`. If you need to clone, you need to manually implement `Clone` to include a call to `aws_endpoints_resolved_endpoint_acquire`.

Reply:
I can implement `Clone`. But, clone was required because I wanted to have a wrapper struct on `ResolvedEndpoint`. But I wanted to know, why do we actually want it?

Reply:
Do we need `Clone` on the wrapper in client?

The reason for the wrapper is that `ResolveEndpoint` is a lower-level type that exposes `get_uri` and `get_properties` which return `OsString`s. In the client we want to return a similar "Resolved Endpoint" type, but which provides methods returning already parsed `Uri` and `AuthScheme` types.